### ISSUE TITLE
Option to disable autoloads

### ISSUE THREAD

**Issue by djrain** [godotengine/godot-proposals#9504]
### Describe the project you are working on

games

### Describe the problem or limitation you are having in your project

When debugging or optimizing I often need to know if something is related to an autoload and wish to disable it. But the options to "disable" an autoload are to either delete it, in which case the game will not run, or to replace it with a dummy script / scene, which is not convenient either

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Add an additional checkbox to each autoload that, if unchecked, the autoload will not be instantiated, but the global variable will remain intact for references. Running code that references a disabled autoload will result in an error.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

just another checkbox

### If this enhancement will not be used often, can it be worked around with a few lines of script?

no

### Is there a reason why this should be core and not an add-on in the asset library?

don't think it's possible

**Comment by KoBeWi** [comment]
You could call `queue_free()` in autoload's `_ready()`.

**Comment by djrain** [comment]
> You could call `queue_free()` in autoload's `_ready()`.

Yeah but it doesn't prevent it from loading. Right now I'm trying to optimize load times so need to avoid all preload etc.

**Comment by KoBeWi** [comment]
Autoloads are very cheap by themselves (unless you put a complex scene as autoload). If you are optimizing startup time, make sure they don't do anything in `_init()`, `_ready()` and other initialization methods. Also don't use preloads in autoload script, because they are loaded with the script at startup (same applies to scripts with `class_name`).

**Comment by djrain** [comment]
Appreciate the tips! I already mostly applied them yet still getting a long splash screen somehow (perhaps related to `class_name` scripts as you mentioned, hadn't thought about those). 

Also consider that addons may not abide by those guidelines and I don't wanna mess with them unless I find that it's a problem. So I still think this feature would have its use.

**Comment by titusio** [comment]
Disabling autoloads would be useful when working with a client/server architecture. It would be nice to just disable autoloads only necessary for each one on export.

**Comment by siarhei-klimuts** [comment]
I am using GodotPlayGameServices addon for android build, which is useless for the steam build and I can ignore the whole addon in export config for steam. But the addon has an autoload which I would like to disable for this export too. So it would be nice to at least have an option to enable/disable autoloads for different export configurations. Maybe using feature flags like in Project Settings.
